services/skus/controllers_noint_test.go

```diff
@@ -222,3 +223,151 @@ func TestHandleReceiptErr(t *testing.T) {
 		})
 	}
 }
+
+func TestParseVerifyCredRequestV2(t *testing.T) {
+	type tcExpected struct {
+		val   *model.VerifyCredentialRequestV2
+		errFn must.ErrorAssertionFunc
+	}
+
+	type testCase struct {
+		name  string
+		given []byte
+		exp   tcExpected
+	}
+
+	tests := []testCase{
+		{
+			name:  "error_malformed_payload",
+			given: []byte(`nonsense`),
+			exp: tcExpected{
+				errFn: func(tt must.TestingT, err error, i ...interface{}) {
+					must.Equal(tt, true, err != nil)
```

Comment:
question: would using `must.NotNil(tt, err)` not be more concise and improve readability?

Reply:
Personally, I do not like tying ourselves to the API of a dependency more and more. This is syntactic sugar, and I don't believe this sugar is necessary. When using a dependency, I prefer using as less as possible.

I've done it in other places before, and those have been passed without a question in past reviews.

My preference would be to stick to the basics unless there is a big win. Using the assertion library is a win, but spreading all of its various methods over the code base does not look appealing.

Happy to discuss it further.

Reply:
This library and its API are used everywhere in the code base and across almost all of our repos including `must.NotNil(tt, err)` so its not clear how adding this convention for asserting on errors is going to alleviate that issue. Would this reasoning also apply to `must.NoError(t, err)` which is prevalent in almost every test we have?

Syntactic sugar is there to make code more readable. Working in the negative and passing in the condition, like above requires more 'mental gymnastics' than simply using the provided API and passing the thing you are asserting on e.g. `must.NotNil(tt, err)`.

Reply:
That's a good point maybe we should updated those tests as well.

This approach was added to the code base along with many other changes to testing which we adopted however, over time this particular feature hasn't proven to add much value hence raising the question now. 

For example, take [this](https://github.com/brave-intl/bat-go/blob/35f7182b31c0b6c7625f4c3359f06df59f4c40d1/services/skus/storage/repository/issuer_test.go#L256) test. There are many steps where we don't want an error before moving on and it is considerably more concise and clear to the reader to simply use the API -

```
must.ErrorIs(t, err, tc.exp.err)

must.NoError(t, err)

```
opposed to whats being advocated above

```
must.Equal(t, true, errors.Is(err, tc.exp.err))

must.Equal(tt, true, err != nil)
```

Reply:
Like I said, I agree with your reasoning, and also have my view on this subject. The choice in this specific PR has been motivated mainly by consistency with what I've done so far.

In one of the future PRs, I will update this and other mentioned tests in accordance with your suggestion as it would allow us to minimise the cognitive load while working with our systems (that already have a fair bit of accidental complexity in them).